surefire-api/src/main/java/org/apache/maven/surefire/api/report/ReporterFactory.java

```diff
@@ -33,7 +33,7 @@
      *
      * @return A reporter instance
      */
-    RunListener createReporter();
+    TestRunListener createReporter();
```

Comment:
rename method `createReporter` -> `crateTestRunListener` WDYT?

javadoc for this method is still actual?

Reply:
In this architecture we have two sides (forked JVM, plugin JVM) having implemented the same interface RunListener:
1. side is ForkingRunListener in the fork JVM
2. TestSetRunListener in the plugin JVM

The serialization between these two JVM ensures that the ReportEntries are transfered from the fork to the plugin.
This is the whole idea behind them.

We can see these renaming issues from the another angle, and so that there is a reporter. Listener means that our surefire `ForkingRunListener` is a delegate within JUnit's RunListeners in the fork. On the other side, the TestSetRunListener is a listener registered in the `ForkClient`.

We can think about it a bit deeper and we will find the best names for these interfaces and methods.
Anyway, we will remove deprecated code, we will create Builders for constructors with too many parameters, so we will have a chance to refactor a lot.

Reply:
@slawekjaranowski yes, the method was renamed in the second commit.